pwiz_tools/Skyline/Controls/Graphs/GraphFullScan.cs

```diff
@@ -1022,11 +1063,18 @@ private void btnIsolationWindow_Click(object sender, EventArgs e)
 
         private void graphControl_ContextMenuBuilder(ZedGraphControl sender, ContextMenuStrip menuStrip, Point mousePt, ZedGraphControl.ContextMenuObjectState objState)
         {
-            ZedGraphHelper.BuildContextMenu(graphControl, menuStrip, true);
+            //ZedGraphHelper.BuildContextMenu(graphControl, menuStrip, true);
+            menuStrip.Tag = this;

```

Comment:
I think it's better to construct a new menu item with an event handler which has all of the variables that it needs, rather than trying to store that information in the "Tag" property of anything.
The problem with using Tag like this is that it's really hard for someone to track down which code is going to use that Tag.

Reply:
There is no clean way to destroy a menu item after it is created. Usually it leads to GDI handles leak, as we discussed. Fortunately, there is a property on the ContextMenuToolStrip pointing to the control that initiated the context menu. I will just use that instead.